src/bluesearch/entrypoint/database/run.py

```diff
@@ -0,0 +1,633 @@
+    )
+    parser.add_argument(
+        "--mesh-topic-db",
+        type=Path,
+        help="""
+        The JSON file with MeSH topic hierarchy information. Mandatory for
+        source types "pmc" and "pubmed".
+
+        The JSON file should contain a flat dictionary with MeSH topic tree
+        numbers mapped to the corresponding topic labels. This file can be
+        produced using the `bbs_database parse-mesh-rdf` command. See that
+        command's description for more details.
+        """,
+    )
+    parser.add_argument(
+        "--dry-run",
+        "-n",
+        action="store_true",
+        help="Prints out a diagram of the pipeline without running it.",
+    )
+    parser.add_argument(
+        "--grobid-host",
+        type=str,
+        help="The host of the GROBID server.",
+    )
+    parser.add_argument(
+        "--grobid-port",
+        type=int,
+        help="The port of the GROBID server.",
+    )
+    parser.add_argument(
+        "--identifier",
+        type=str,
+        help="""Custom name of the identifier. If not specified, we use
+        `from-month_today`
+        """,
+    )
+    parser.add_argument(
+        "--final-task",
+        type=str,
+        help="""Name of the task where to manually stop the pipeline. Note
+        that the task itself will be included.
+        """,
+    )
+
+    return parser
+
+
+BBS_BINARY = ["bbs_database"]
+VERBOSITY = ["-v"]  # for the entrypoint subprocesses
+CAPTURE_OUTPUT = False
+IDENTIFIER = None  # make sure the same for all tasks
+
+
+class DownloadTask(ExternalProgramTask):
+    """Download raw files.
+
+    They will be stored in the `raw/` folder.
+    """
+
+    source = luigi.Parameter()
+    from_month = luigi.Parameter()
+    output_dir = luigi.Parameter()
+    identifier = luigi.OptionalParameter()
+
+    def output(self) -> luigi.LocalTarget:
+        """Define download folder."""
+        global IDENTIFIER
+        if self.identifier is not None:
+            identifier = self.identifier
+
+        else:
+            if IDENTIFIER is None:
+                today = datetime.today()
+                identifier = f"{self.from_month}_{today.strftime('%Y-%m-%d')}"
+                IDENTIFIER = identifier
+            else:
+                identifier = IDENTIFIER
+
+        output_dir = Path(self.output_dir) / self.source / identifier / "raw"
+
+        return luigi.LocalTarget(str(output_dir))
+
+    def program_args(self) -> list[str]:
+        """Define subprocess arguments."""
+        output_dir = self.output().path
+        return [
+            *BBS_BINARY,
+            "download",
+            *VERBOSITY,
+            self.source,
+            self.from_month,
+            output_dir,
+        ]
+
+
+@requires(DownloadTask)
+class UnzipTask(ExternalProgramTask):
+    """Unzip raw files (if necessary).
+
+    Only applicable in case of `pmc`. The unzipped files
+    are stored inside of `raw_unzipped`.
+    """
+
+    source = luigi.Parameter()
```

Comment:
Don't we already have
```python
@requires(DownloadTask)
```
a few lines before?

Reply:
I am not sure what you are referring to, however, `source` is going to be `arxiv`, `medrix`, etc

Reply:
Ahh, I think I get your point now. Yes, the `requires` is going copy paste all the parameters of the `DownloadTask`. So it is not necessary to specify it again, however, we just though it would be nice to define it explicitly again those parameters that are necessary for that given task. But it is definitely something we can discuss.